### PR TITLE
Implement Conduit endpoints

### PR DESCRIPTION
**Pull request by melon095**
This PR implements the six new Helix endpoints for conduits.

I want to know your opinion on [the conduit transport converter](https://github.com/melon095/MiniTwitch/blob/49427852753ba350d5c695520d531fe6bbffc9a0/MiniTwitch.Helix/Responses/ConduitTransport.cs#L15-L36). I wanted to use this hacky discriminated union since it feels the nicest to use. But the _JsonConverter_ that comes with it isn't the nicest. I know that in .NET 7 we got _JsonDerivedType_ but since this library compiles to .NET 6 i can't use that.

Updating it should be as simple as adding this
```patch
+[JsonPolymorphic(TypeDiscriminatorPropertyName = "method")]
+[JsonDerivedType(typeof(WebSocket), typeDiscriminator: "websocket")]
+[JsonDerivedType(typeof(Webhook), typeDiscriminator: "webhook")]
-[JsonConverter(typeof(ConduitTransportConverter))]
```

### CODE REVIEW

File: MiniTwitch.Helix/Responses/ConduitShard.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿using System.Text.Json.Serialization;
+using MiniTwitch.Helix.Enums;
+using MiniTwitch.Helix.Internal.Json;
+
+namespace MiniTwitch.Helix.Responses;
+
+public record ConduitShard(
```
Comment (Foretack) [inline MiniTwitch.Helix/Responses/ConduitShard.cs:7]: This should be nested under ConduitsShards, see Responses/Chatters.cs as an example

File: MiniTwitch.Helix/Responses/ConduitTransport.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+[JsonConverter(typeof(ConduitTransportConverter))]
+public abstract record ConduitTransport()
+{
+    public string Method { get; }
+
+    public sealed record WebSocket(string SessionId, DateTime ConnectedAt, DateTime DisconnectedAt) : ConduitTransport();
+    public sealed record Webhook(string Callback) : ConduitTransport();
+}
+
+public class ConduitTransportConverter : JsonConverter<ConduitTransport>
```
Comment (Foretack) [inline MiniTwitch.Helix/Responses/ConduitTransport.cs:15]: Move this to Internal/Json/Converters.cs (or its own file)

**Review by Foretack — CHANGES_REQUESTED** [review]

**Comment by Foretack** [comment]
> I want to know your opinion on [the conduit transport converter](https://github.com/melon095/MiniTwitch/blob/49427852753ba350d5c695520d531fe6bbffc9a0/MiniTwitch.Helix/Responses/ConduitTransport.cs#L15-L36). I wanted to use this hacky discriminated union since it feels the nicest to use. But the _JsonConverter_ that comes with it isn't the nicest. I know that in .NET 7 we got _JsonDerivedType_ but since this library compiles to .NET 6 i can't use that.

Built in json has a lot of problems because of net6 support, I'm a big fan of the net7+ approach, and I'd be happy to switch to it once support is dropped for net6. Thanks for bringing this up 👍

File: MiniTwitch.Helix/Responses/ConduitShard.cs (hunk 1 above)
Comment (melon095) [reply MiniTwitch.Helix/Responses/ConduitShard.cs]: Fixed in 2e4a6b9ae27f337aade7ceeb56fddef351ab1c5b

File: MiniTwitch.Helix/Responses/ConduitTransport.cs (hunk 2 above)
Comment (melon095) [reply MiniTwitch.Helix/Responses/ConduitTransport.cs]: Fixed in 2e4a6b9ae27f337aade7ceeb56fddef351ab1c5b

**Review by Foretack — APPROVED** [review]
LGTM, I'll go through it again tomorrow and I'll merge if I don't have any comments

**Comment by Foretack** [comment]
Works as expected. Thank you for your contribution ❤️

**Pull request merged**